### DISCUSSION TITLE
Everyday Usage

### DISCUSSION THREAD

**Discussion by falavu** [Homebrew/discussions#5545] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.13
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1e560120a65be3b3c7b8a42083d54f795e1d0ec4
Last commit: 30 hours ago
Core tap JSON: 06 Aug 13:35 UTC
Core cask tap JSON: 06 Aug 13:35 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.aIVlPVMcAn/org.xquartz:0
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 3.3.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: quad-core 64-bit haswell
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 12.7.6-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
```

### Output of `brew doctor`

```shell
Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  jam
  python-packaging
  trimage

Warning: libiconv files detected at a system prefix other than /usr.
Homebrew doesn't provide a libiconv formula and expects to link against
the system version in /usr. libiconv in other prefixes can cause
compile or link failure, especially if compiled with improper
architectures. macOS itself never installs anything to /usr/local so
it was either installed by a user or some other third party software.

tl;dr: delete these files:
  /usr/local/lib/libiconv.dylib
  /usr/local/include/iconv.h
  /usr/local/lib/libiconv.dylib
  /usr/local/include/iconv.h

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libblkid.2.0.dylib
  /usr/local/lib/libcharset.1.0.0.dylib
  /usr/local/lib/libcom_err.1.1.dylib
  /usr/local/lib/libdmtx.0.dylib
  /usr/local/lib/libe2p.2.1.dylib
  /usr/local/lib/libexif.12.dylib
  /usr/local/lib/libexpat.1.dylib
  /usr/local/lib/libext2fs.2.1.dylib
  /usr/local/lib/libgcrypt.11.5.2.dylib
  /usr/local/lib/libgdal.1.dylib
  /usr/local/lib/libiconv.2.2.0.dylib
  /usr/local/lib/libiconv.2.4.0.dylib
  /usr/local/lib/libjasper.1.0.0.dylib
  /usr/local/lib/libjson-c.2.dylib
  /usr/local/lib/libproj.0.dylib
  /usr/local/lib/libss.1.0.dylib
  /usr/local/lib/libusb-0.1.4.4.4.dylib
  /usr/local/lib/libusbpp-0.1.4.4.4.dylib
  /usr/local/lib/libuuid.1.1.dylib
  /usr/local/lib/libykclient.3.dylib
  /usr/local/lib/libykpers-1.1.dylib
  /usr/local/lib/libyubikey.0.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected header files:
  /usr/local/include/Magick++.h
  /usr/local/include/Magick++/Blob.h
  /usr/local/include/Magick++/CoderInfo.h
  /usr/local/include/Magick++/Color.h
  /usr/local/include/Magick++/Drawable.h
  /usr/local/include/Magick++/Exception.h
  /usr/local/include/Magick++/Geometry.h
  /usr/local/include/Magick++/Image.h
  /usr/local/include/Magick++/Include.h
  /usr/local/include/Magick++/Montage.h
  /usr/local/include/Magick++/Pixels.h
  /usr/local/include/Magick++/STL.h
  /usr/local/include/Magick++/TypeMetric.h
  /usr/local/include/cpl_atomic_ops.h
  /usr/local/include/cpl_config.h
  /usr/local/include/cpl_config_extras.h
  /usr/local/include/cpl_conv.h
  /usr/local/include/cpl_csv.h
  /usr/local/include/cpl_error.h
  /usr/local/include/cpl_hash_set.h
  /usr/local/include/cpl_http.h
  /usr/local/include/cpl_list.h
  /usr/local/include/cpl_minixml.h
  /usr/local/include/cpl_minizip_ioapi.h
  /usr/local/include/cpl_minizip_unzip.h
  /usr/local/include/cpl_minizip_zip.h
  /usr/local/include/cpl_multiproc.h
  /usr/local/include/cpl_odbc.h
  /usr/local/include/cpl_port.h
  /usr/local/include/cpl_quad_tree.h
  /usr/local/include/cpl_string.h
  /usr/local/include/cpl_time.h
  /usr/local/include/cpl_vsi.h
  /usr/local/include/cpl_vsi_virtual.h
  /usr/local/include/cpl_win32ce_api.h
  /usr/local/include/cpl_wince.h
  /usr/local/include/cplkeywordparser.h
  /usr/local/include/dmtx.h
  /usr/local/include/expat.h
  /usr/local/include/expat_external.h
  /usr/local/include/gdal.h
  /usr/local/include/gdal_alg.h
  /usr/local/include/gdal_alg_priv.h
  /usr/local/include/gdal_csv.h
  /usr/local/include/gdal_frmts.h
  /usr/local/include/gdal_pam.h
  /usr/local/include/gdal_priv.h
  /usr/local/include/gdal_proxy.h
  /usr/local/include/gdal_rat.h
  /usr/local/include/gdal_version.h
  /usr/local/include/gdal_vrt.h
  /usr/local/include/gdalgrid.h
  /usr/local/include/gdaljp2metadata.h
  /usr/local/include/gdalwarper.h
  /usr/local/include/gdalwarpkernel_opencl.h
  /usr/local/include/gvgcpfit.h
  /usr/local/include/iconv.h
  /usr/local/include/jasper/jas_cm.h
  /usr/local/include/jasper/jas_config.h
  /usr/local/include/jasper/jas_config2.h
  /usr/local/include/jasper/jas_debug.h
  /usr/local/include/jasper/jas_fix.h
  /usr/local/include/jasper/jas_getopt.h
  /usr/local/include/jasper/jas_icc.h
  /usr/local/include/jasper/jas_image.h
  /usr/local/include/jasper/jas_init.h
  /usr/local/include/jasper/jas_malloc.h
  /usr/local/include/jasper/jas_math.h
  /usr/local/include/jasper/jas_seq.h
  /usr/local/include/jasper/jas_stream.h
  /usr/local/include/jasper/jas_string.h
  /usr/local/include/jasper/jas_tmr.h
  /usr/local/include/jasper/jas_tvp.h
  /usr/local/include/jasper/jas_types.h
  /usr/local/include/jasper/jas_version.h
  /usr/local/include/jasper/jasper.h
  /usr/local/include/libcharset.h
  /usr/local/include/libexif/_stdint.h
  /usr/local/include/libexif/exif-byte-order.h
  /usr/local/include/libexif/exif-content.h
  /usr/local/include/libexif/exif-data-type.h
  /usr/local/include/libexif/exif-data.h
  /usr/local/include/libexif/exif-entry.h
  /usr/local/include/libexif/exif-format.h
  /usr/local/include/libexif/exif-ifd.h
  /usr/local/include/libexif/exif-loader.h
  /usr/local/include/libexif/exif-log.h
  /usr/local/include/libexif/exif-mem.h
  /usr/local/include/libexif/exif-mnote-data.h
  /usr/local/include/libexif/exif-tag.h
  /usr/local/include/libexif/exif-utils.h
  /usr/local/include/localcharset.h
  /usr/local/include/magick/ImageMagick.h
  /usr/local/include/magick/PreRvIcccm.h
  /usr/local/include/magick/annotate.h
  /usr/local/include/magick/api.h
  /usr/local/include/magick/attribute.h
  /usr/local/include/magick/blob.h
  /usr/local/include/magick/cache.h
  /usr/local/include/magick/cache_view.h
  /usr/local/include/magick/client.h
  /usr/local/include/magick/coder.h
  /usr/local/include/magick/color.h
  /usr/local/include/magick/colorspace.h
  /usr/local/include/magick/compare.h
  /usr/local/include/magick/composite.h
  /usr/local/include/magick/compress.h
  /usr/local/include/magick/configure.h
  /usr/local/include/magick/conjure.h
  /usr/local/include/magick/constitute.h
  /usr/local/include/magick/convert.h
  /usr/local/include/magick/decorate.h
  /usr/local/include/magick/delegate.h
  /usr/local/include/magick/deprecate.h
  /usr/local/include/magick/draw.h
  /usr/local/include/magick/effect.h
  /usr/local/include/magick/enhance.h
  /usr/local/include/magick/exception.h
  /usr/local/include/magick/fx.h
  /usr/local/include/magick/gem.h
  /usr/local/include/magick/geometry.h
  /usr/local/include/magick/hashmap.h
  /usr/local/include/magick/identify.h
  /usr/local/include/magick/image.h
  /usr/local/include/magick/import.h
  /usr/local/include/magick/list.h
  /usr/local/include/magick/locale_.h
  /usr/local/include/magick/log.h
  /usr/local/include/magick/magic.h
  /usr/local/include/magick/magick.h
  /usr/local/include/magick/magick_config.h
  /usr/local/include/magick/memory_.h
  /usr/local/include/magick/methods.h
  /usr/local/include/magick/module.h
  /usr/local/include/magick/mogrify.h
  /usr/local/include/magick/monitor.h
  /usr/local/include/magick/montage.h
  /usr/local/include/magick/option.h
  /usr/local/include/magick/paint.h
  /usr/local/include/magick/profile.h
  /usr/local/include/magick/quantize.h
  /usr/local/include/magick/random.h
  /usr/local/include/magick/registry.h
  /usr/local/include/magick/resize.h
  /usr/local/include/magick/resource_.h
  /usr/local/include/magick/segment.h
  /usr/local/include/magick/semaphore.h
  /usr/local/include/magick/shear.h
  /usr/local/include/magick/signature.h
  /usr/local/include/magick/stream.h
  /usr/local/include/magick/string_.h
  /usr/local/include/magick/timer.h
  /usr/local/include/magick/token.h
  /usr/local/include/magick/transform.h
  /usr/local/include/magick/type.h
  /usr/local/include/magick/utility.h
  /usr/local/include/magick/version.h
  /usr/local/include/magick/widget.h
  /usr/local/include/magick/xwindow.h
  /usr/local/include/memdataset.h
  /usr/local/include/neon/ne_207.h
  /usr/local/include/neon/ne_acl.h
  /usr/local/include/neon/ne_alloc.h
  /usr/local/include/neon/ne_auth.h
  /usr/local/include/neon/ne_basic.h
  /usr/local/include/neon/ne_compress.h
  /usr/local/include/neon/ne_dates.h
  /usr/local/include/neon/ne_defs.h
  /usr/local/include/neon/ne_i18n.h
  /usr/local/include/neon/ne_locks.h
  /usr/local/include/neon/ne_md5.h
  /usr/local/include/neon/ne_props.h
  /usr/local/include/neon/ne_redirect.h
  /usr/local/include/neon/ne_request.h
  /usr/local/include/neon/ne_session.h
  /usr/local/include/neon/ne_socket.h
  /usr/local/include/neon/ne_ssl.h
  /usr/local/include/neon/ne_string.h
  /usr/local/include/neon/ne_uri.h
  /usr/local/include/neon/ne_utils.h
  /usr/local/include/neon/ne_xml.h
  /usr/local/include/neon/ne_xmlreq.h
  /usr/local/include/ogr_api.h
  /usr/local/include/ogr_core.h
  /usr/local/include/ogr_feature.h
  /usr/local/include/ogr_featurestyle.h
  /usr/local/include/ogr_geometry.h
  /usr/local/include/ogr_p.h
  /usr/local/include/ogr_spatialref.h
  /usr/local/include/ogr_srs_api.h
  /usr/local/include/ogrsf_frmts.h
  /usr/local/include/org_proj4_Projections.h
  /usr/local/include/proj_api.h
  /usr/local/include/rawdataset.h
  /usr/local/include/subversion-1/mod_dav_svn.h
  /usr/local/include/subversion-1/svn_auth.h
  /usr/local/include/subversion-1/svn_base64.h
  /usr/local/include/subversion-1/svn_client.h
  /usr/local/include/subversion-1/svn_cmdline.h
  /usr/local/include/subversion-1/svn_config.h
  /usr/local/include/subversion-1/svn_ctype.h
  /usr/local/include/subversion-1/svn_dav.h
  /usr/local/include/subversion-1/svn_delta.h
  /usr/local/include/subversion-1/svn_diff.h
  /usr/local/include/subversion-1/svn_error.h
  /usr/local/include/subversion-1/svn_error_codes.h
  /usr/local/include/subversion-1/svn_fs.h
  /usr/local/include/subversion-1/svn_hash.h
  /usr/local/include/subversion-1/svn_io.h
  /usr/local/include/subversion-1/svn_md5.h
  /usr/local/include/subversion-1/svn_nls.h
  /usr/local/include/subversion-1/svn_opt.h
  /usr/local/include/subversion-1/svn_path.h
  /usr/local/include/subversion-1/svn_pools.h
  /usr/local/include/subversion-1/svn_props.h
  /usr/local/include/subversion-1/svn_quoprint.h
  /usr/local/include/subversion-1/svn_ra.h
  /usr/local/include/subversion-1/svn_ra_svn.h
  /usr/local/include/subversion-1/svn_repos.h
  /usr/local/include/subversion-1/svn_sorts.h
  /usr/local/include/subversion-1/svn_string.h
  /usr/local/include/subversion-1/svn_subst.h
  /usr/local/include/subversion-1/svn_time.h
  /usr/local/include/subversion-1/svn_types.h
  /usr/local/include/subversion-1/svn_utf.h
  /usr/local/include/subversion-1/svn_version.h
  /usr/local/include/subversion-1/svn_wc.h
  /usr/local/include/subversion-1/svn_xml.h
  /usr/local/include/thinplatespline.h
  /usr/local/include/vrtdataset.h
  /usr/local/include/wand/drawing_wand.h
  /usr/local/include/wand/magick_wand.h
  /usr/local/include/wand/pixel_iterator.h
  /usr/local/include/wand/pixel_wand.h

Warning: Unbrewed '.la' files were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected '.la' files:
  /usr/local/lib/libcharset.la
  /usr/local/lib/libdmtx.la
  /usr/local/lib/libexif.la
  /usr/local/lib/libexpat.la
  /usr/local/lib/libgdal.la
  /usr/local/lib/libiconv.la
  /usr/local/lib/libjasper.la
  /usr/local/lib/libjpeg.la
  /usr/local/lib/libproj.la
  /usr/local/lib/libusbpp.la

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libcharset.a

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  ghostscript
  libexif
  python@3.12
  rust
  libgpg-error
  handbrake
  python@3.11

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc

Warning: Broken symlinks were found. Remove them with `brew cleanup`:
  /usr/local/bin/appdb
  /usr/local/bin/github
  /usr/local/bin/msiexec
  /usr/local/bin/notepad
  /usr/local/bin/regedit
  /usr/local/bin/regsvr32
  /usr/local/bin/wine
  /usr/local/bin/wine-preloader
  /usr/local/bin/wine64
  /usr/local/bin/wine64-preloader
  /usr/local/bin/wineboot
  /usr/local/bin/winecfg
  /usr/local/bin/wineconsole
  /usr/local/bin/winefile
  /usr/local/bin/winehelp
  /usr/local/bin/winemine
  /usr/local/bin/winepath
  /usr/local/bin/wineserver
  /usr/local/opt/rust
```

### Description of issue

Hello, 
i am a normal user, having less skills on the Operating System macOS (12.7), but i use Homebrew for Interesting on the Concept. 
When i „brew upgrade“, i get this: Error: /usr/local/Cellar/rust/1.79.0_1 is not a directory
It appers suddenly – i dont know have done something on the system. 
Please sorry for my cruel English (with Help from Deepl).
Thank You very much for helping me.

**Comment by yushli** [comment, marked as answer]
Running brew cleanup fixed the issue for me.
